MyBus/MyBus/MyBusMapController.swift

```diff
@@ -344,32 +348,39 @@ class MyBusMapController: UIViewController, MGLMapViewDelegate, UITableViewDeleg
             SearchManager.sharedInstance.getRoad(route) {
                 road, error in
                 self.progressNotification.stopLoadingNotification(self.view)
-                if let routeRoad = road {                    
+                if let routeRoad = road {
                     self.updateRoad(routeRoad)
                 } else {
                     GenerateMessageAlert.generateAlert(self, title: "Tuvimos un problema 😿", message: "No pudimos resolver el detalle de la opción seleccionada")
                 }
             }
         }
     }
-    
-    
+
+
     //Newest implementation
-    func updateOrigin(newOrigin:RoutePoint?){
+    func updateOrigin(newOrigin: RoutePoint?){
         self.mapView.clearExistingBusRouteAnnotations()
-        
+        self.mapView.clearExistingBusRoadAnnotations()
+        resetHideResultsTable()
         if let origin = newOrigin {
             let marker = MyBusMarkerFactory.createOriginPointMarker(origin)
             self.mapModel.originMarker = marker
         }else{
             self.mapView.removeOriginPoint()
             self.mapModel.originMarker = nil
         }
+
+        //If destination is set, we should fit to origin and destionation marker
+        if let _ = self.mapModel.destinationMarker {
+            self.mapView.fitToAnnotationsInMap()
+        }
     }
-    
-    func updateDestination(newDestination:RoutePoint?){
+
+    func updateDestination(newDestination: RoutePoint?){
         self.mapView.clearExistingBusRouteAnnotations()
-        
+        self.mapView.clearExistingBusRoadAnnotations()
```

Comment:
If I understand right, I believe that the behaviour that you mention has already been covered in line 381.